exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -525,6 +526,40 @@ interface ISqlExpressionBuilder {
         caseSensitive: Boolean = true
     ): RegexpOp<T> = RegexpOp(this, pattern, caseSensitive)
 
+    // JSON Conditions
+
+    /**
+     * Checks whether a [candidate] expression is contained within [this] JSON expression.
+     *
+     * @param candidate Expression to search for in [this] JSON expression.
+     * @param path String representing JSON path/keys that match specific fields to search for [candidate].
+     * **Note:** Optional [path] argument is not supported by all vendors; please check the documentation.
+     */
+    fun ExpressionWithColumnType<*>.jsonContains(candidate: Expression<*>, path: String? = null): JsonContains =
+        JsonContains(this, candidate, path, columnType)
+
+    /**
+     * Checks whether a [candidate] value is contained within [this] JSON expression.
+     *
+     * @param candidate Value to search for in [this] JSON expression.
+     * @param path String representing JSON path/keys that match specific fields to search for [candidate].
+     * **Note:** Optional [path] argument is not supported by all vendors; please check the documentation.
+     */
+    fun <T> ExpressionWithColumnType<*>.jsonContains(candidate: T, path: String? = null): JsonContains =
+        JsonContains(this, asLiteral(candidate), path, columnType)
```

Comment:
Having 2 overloads means the function can be called with a primitive type value, like a String, or an expression, like another json column or the result of another json function, for example.

I considered a 3rd overload, if the user wants to pass a data class instance as `candidate` that gets serialized to a String, something like:
```kt
inline fun <reified T : Any> ExpressionWithColumnType<*>.jsonContains(candidate: T, jsonConfig: Json, kSerializer: KSerializer<T> = serializer<T>(), path: String? = null): JsonContains =
        JsonContains(this, jsonConfig.encodeToString(kSerializer, it), path, columnType)
```
but it doesn't work in an interface (`SqlExpressionBuilder`) and I'm not very sure about the use case given [the rules of containment and nesting](https://www.postgresql.org/docs/15/datatype-json.html#JSON-CONTAINMENT) (also [in MySQL](https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#function_json-contains)). So I'm tempted to bench this 3rd overload unless it's requested and, hopefully, a more concrete use case is provided that can be tested.

Some examples of this are in `testJsonContains()`.